src/canvas/Canvas.ts

```diff
@@ -917,34 +913,29 @@ export class Canvas extends SelectableCanvas implements CanvasOptions {
   /**
    * @private
    * Handle event firing for target and subtargets
-   * @param {Event} e event from mouse
-   * @param {String} eventType event to fire (up, down or move)
-   * @param {fabric.Object} targetObj receiving event
+   * @param {TPointerEvent} e event from mouse
+   * @param {TPointerEventNames} eventType
    * @param {Number} [button] button used in the event 1 = left, 2 = middle, 3 = right
-   * @param {Boolean} isClick for left button only, indicates that the mouse up happened without move.
    */
-  _handleEvent(
+  _handleEvent<T extends TPointerEventNames>(
     e: TPointerEvent,
-    eventType: TPointerEventNames,
-    button = LEFT_CLICK,
-    isClick = false
+    eventType: T,
+    button = LEFT_CLICK
   ) {
     const target = this._target,
       targets = this.targets || [],
-      options: TPointerEventInfo = {
-        e: e,
-        target: target,
+      options: CanvasEvents[`mouse:${T}`] = {
+        e,
+        target,
         subTargets: targets,
         button,
-        isClick,
         pointer: this.getPointer(e, true),
         absolutePointer: this.getPointer(e),
         transform: this._currentTransform,
-      };
-    if (eventType === 'up') {
-      options.currentTarget = this.findTarget(e);
-      options.currentSubTargets = this.targets;
```

Comment:
I have no idea why this was like this
Nothing is using it in fabric so I removed it
Any objections?
The dev can call `findTarget` from an event handler if they need

Reply:
i would rather keep this, is kind of unrelated to the PR and i don't have the tools or the time into digging why this is here.
There is for sure some connection with the current transform that return a different target from the one that actually is possibly under the mouse, and untill today we surfaced it and patched it in this way.